File: demo/src/pages/TransformView/TransformView.swift
//

import SwiftUI

struct TransformView: View {
    @State private var angle = 0.0;
    var body: some View {
        Text("Rotate Me!")
            .rotationEffect(.degrees(angle))
        Slider(value:$angle, in:0...360)
    }
}

struct TransformView_Previews: PreviewProvider {
    static var previews: some View {
        TransformView()
    }
}

File: demo/src/pages/TransformView/TransformView.ts
import { Bindable, State, degrees, Viewable, Slider, Text, View, Bound, Num, main } from "@tswift/ui";

@main
export class TransformView extends Viewable {
  @State angle: Num = 0;

  body = ({ $angle, angle }: Bound<this>): View[] => {
    return [Text("Rotate Me").rotationEffect(degrees(angle), ".center"), Slider({ value: $angle, in: "0...360" })];
  };
}
